### PR TITLE
Booster

### PR DESCRIPTION
**Pull request by Matt-BF**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by Matt-BF** [comment]
@bioconda/core @bioconda/osx I would like some help with building this recipe for osx. I’ve tried a few workarounds with clang but so far haven’t gained much ground. Any suggestions that may solve this problem? Thanks in advance!

File: recipes/booster/build.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+CXXFLAGS="$CXXFLAGS";
+CFLAGS="$CFLAGS";
+LDFLAGS="$LDFLAGS -Wl,-rpath ${PREFIX}/lib";
+CXX=g++;
+CC=gcc;
```
Comment (dpryan79) [inline recipes/booster/build.sh]: Please don't change CXX or CC.

File: recipes/booster/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  url: https://github.com/evolbioinfo/booster/archive/v{{ version }}.tar.gz
+  sha256: eee92b91c98d3fccb2b94201dfdc775d1c381d22c5953240562b8de20e763e18
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - llvmdev  # [osx]
+    - libcxx 4.0.*  # [osx]
```
Comment (dpryan79) [inline recipes/booster/meta.yaml]: try adding `llvm-openmp` to the `build:` and `run:` sections. You can then remove `openmp` (actually, you should be able to remove the entirety of what's now in your `host:` and `run:` sections).

File: recipes/booster/build.sh (hunk 1 above)
Comment (Matt-BF) [reply recipes/booster/build.sh]: > If you can get `make` to respect the compilers specified by `CC` and `CXX` then the builds should work.

Could you elaborate more on that? I've no experience with C programs. Also, I added your suggestions to the `meta.yaml` file and rolled back the `build.sh` file to a simple `make` call.

Comment (dpryan79) [reply recipes/booster/build.sh]: It's mostly a matter of looking into the Makefile and seeing how it's defining which compiler to use. In many cases, it's defining a variable like `${CC}`, which makes life easy. In other cases the compiler may be hard-coded (e.g., `gcc -o foo foo.c`), which means you'll have to heavily modify the Makefile so it uses the correct compilers. I'll try to take a look at the make file used by this project and see if I can provide any pointers.

**Comment by dpryan79** [comment]
Looks like the changes I made should suffice.

**Comment by Matt-BF** [comment]
@dpryan79 wow! Thank you very much, I'm gonna read carefully what you did and see if I might learn something for next time!

**Pull request merged**